share/spack/qa/no_changes_in_core.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env spack-python
+#
+# Description:
+#     Helper script that returns 0 if any file in core
+#     has been changed, 1 otherwise
+#
+# Usage:
+#     nochanges_in_core.py && [command run if there are no changes in core]
+#
+
+import sys
+import os.path
+
+import spack.spec
+import spack.cmd.flake8
+
+# Get the complete list of files that changed
+try:
+    files = spack.cmd.flake8.changed_files(base='develop', untracked=True)
+except Exception:
+    print('ERROR: cannot compute the list of files that changed.')
+    sys.exit(1)
+
+# If something changed in the core libraries we need to test it
+core_path = os.path.join('lib', 'spack')
+changes_in_core = any(core_path in x for x in files)
+
+# Exit early because polling the repo may be slow
+if changes_in_core:
+    print('FAILURE: at least one core file has been modified.')
+    print('Rename your branch to something that does not start with "packages"')  # noqa: ignore=E501
```

Comment:
Just a quick comment on this. The only failure case should be if somebody submitted a change to the core framework and called the branch `packages/*`. For the most common case where somebody changed just a package and named the branch whatever they want, that just triggers all the tests. It will be slower, but won't result in a failure because of naming.

:+1: to the suggestion of sending an automated message to tell people they had a faster way to run tests, if they missed it.